app-typescript/components/DateTimePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {DatePicker} from '../components/DatePicker';
+import {Spacer} from '@superdesk/common';
+import {defaultTo, padStart} from 'lodash';
+import {TimePicker} from './TimePicker';
+import {IconButton} from './IconButton';
+
+interface IProps {
+    value: Date | null;
+    label: string;
+    dateFormat: string;
+    onChange: (value?: string) => void;
+    preview?: boolean;
+    fullWidth?: boolean;
+    allowSeconds?: boolean;
+    required?: boolean;
+    width?: React.CSSProperties['width'];
+    disabled?: boolean;
+}
+
+const MIN_WIDTH = 348;
+
+export class DateTimePicker extends React.PureComponent<IProps> {
+    handleTimeChange = (time: string) => {
+        const [hours, minutes] = time.split(':').map((x) => defaultTo(parseInt(x, 10), 0)); // handle NaN value
```

Comment:
FYI there's also `isNaN` function

Reply:
I'm aware. But in the case of using isNaN the whole expression becomes less readable in my opinion, because isNaN only checks whether the value is `NaN`, `defaultTo` does the same but also offers an immediate fallback. So in this case it is a better fit. Otherwise if you prefer a more explicit style, with isNaN it would look like `isNan(x) ? 0 : parseInt(x, 10)`.

Reply:
You can leave it as you prefer, I don't mind. I just wanted to point that `isNan` function exists. Personally I slightly prefer `isNan` approach, because it doesn't use libraries and person reading the code doesn't need to understand what the library function does. I dislike lodash specifically for making the same function compatible with many data types. It's good on one side, but on another, it's making it harder to look at code and infer from usage what's going on.